gulpfile.js

```diff
@@ -32,12 +33,21 @@ gulp.task('serve-pattern-library', () => {
 // The following tasks bundle assets for the pattern library for use locally
 // during development. Bundled JS and CSS are not published with the package.
 
-gulp.task('bundle-css', () => buildCSS(['./styles/pattern-library.scss']));
+gulp.task('bundle-css', () =>
+  buildCSS(['./styles/pattern-library.scss'], { tailwindConfig })
+);
 
 gulp.task(
   'watch-css',
   gulp.series('bundle-css', () =>
-    gulp.watch('./styles/**/*.scss', gulp.task('bundle-css'))
+    gulp.watch(
+      [
+        './styles/**/*.scss',
+        './src/components/**/*.js',
+        './src/pattern-library/**/*.js',
```

Comment:
Don't we use Tailwind JIT compiler, shouldn't these files be watched out of the box? How does Gulp/Tailwind (and sometimes SASS) play all together?

Reply:
@esanzgar 

I can see the source of confusion here! Let me see if I can explain the distinction between _watching_ (what these globs define) and _purging_ (globs defined in the `tailwind.config`'s `content` property).

When pre-processing CSS source, Tailwind will only add classes to the output CSS stylesheet(s) that are referenced within the files defined in the [configuration's `content` glob(s)](https://github.com/hypothesis/frontend-shared/blob/b10f543b5697cdcc9e3c0fd9ac833a774089a413/tailwind.config.js#L5). It will not add any of its utility classes[^1], per its JIT strategy, unless it "sees" a class in a source `content` file. So this `content` property in the Tailwind configuration is how we tell Tailwind *where to look* when it's building.

Secondarily, we need to make sure our CSS watching *rebuilds at the right times* to "tell" Tailwind that it should take another look at its `content` sources for possible changes to classes being used. In the past, all of the information needed to build CSS was contained within the source SCSS files. Now we have a situation in which changes to other source have bearing on the generated CSS. Say you edit the `MyWidget` component and add a `bg-grey-4` class to an element in it. We need to re-build CSS to make sure that Tailwind "saw" that the `bg-grey-4` utility class it being used and make it available in the output stylesheet.

So I guess I'd say the `content` prop in the Tailwind config is the "where" and these changes to the watch answer the "when", as it were.

Does this make sense?

[^1]: There are ways to configure Tailwind to make sure specific classes are always available using [safelisting](https://tailwindcss.com/docs/content-configuration#safelisting-classes)

Reply:
> If tailwindcss/base is included but the preflight is disable, what rules are included? None?

@esanzgar It's a bit subtle and I don't think Tailwind's docs do a great job of explaining the role of the `base` layer. Its role of applying base styles — CSS normalize and reset — is generally apparent, but we're suppressing that by disabling the `preflight` plugin. The other role it has is setting up some root-level CSS variables that are necessary for certain transforms, animations, etc. It's one of those things that you don't realize you need until some utility class just isn't working right.

I expect to re-enable `preflight` within the next short while once I have an hour or so to spare to refactor some pattern-library CSS (I look forward to it).